packages/uploadthing/src/client.ts

```diff
@@ -183,9 +187,25 @@ export const DANGEROUS__uploadFiles = async <TRouter extends FileRouter>(
     );
 
     if (upload.status > 299 || upload.status < 200) {
+      // tell uploadthing infra server that upload failed
+      await fetch(
+        createRequestUrl({
+          url: config?.url,
+          slug: String(opts.endpoint),
+          actionType: "failure",
+        }),
+        {
+          method: "POST",
+          body: JSON.stringify({
+            fileKey: fields.key,
+          }),
+        },
+      );
```

Comment:
afaik this calls your server first, and then that will call uploadthing infra? is the plan to add an `onUploadError` hook in the file route config which will make this step necessary, or can we just send the fetch directly to uploadthing?

Reply:
would it maybe also make sense to throw here instead, and then in an `.allSettled` check for errors and send a single request to the server, instead of 1 per file?

Reply:
Yeah, I am not sure why I went the route of going through your server first tbh...

Reply:
Right, I did this because we don't have the API key on client.